YoutubeExplode/Playlists/PlaylistClient.cs

```diff
@@ -92,8 +92,12 @@ public async IAsyncEnumerable<Batch<PlaylistVideo>> GetVideoBatchesAsync(
             );
 
             var videos = new List<PlaylistVideo>();
+            var originalVideos = response.Videos.Where(v =>
+            {
+                return !encounteredIds.Any(e => string.Equals(e.Value, v.Id));
+            });
 
-            foreach (var videoData in response.Videos)
+            foreach (var videoData in originalVideos)
```

Comment:
> I'm confused what fundamentally changed between the old and the current implementation. Can you explain in more detail?

Sorry for late answer. To be honest it was hard for me to detect it (and even harder to explain it), but I noticed that sometimes our duplicates can be as a last list element and while they are not added to `encounteredId`, they are still used in JSON key `playlistIndex`. That's why when we are reaching value of 2039 in `encounteredId`, we have value 811 of `lastVideoIndex`.  So, to be sure that we don't work with duplicates, we can remove them before foreach.  

It's probably make sense to remove if condition in foreach and just add `videoId` to `encounteredId`, since we removing duplicates.

Reply:
BTW, I realised that duplicates might be in the beginning and then my suggestion doesn't cover it, so I will improve solution.